### ISSUE TITLE
v1.0.2 won't compile under Xcode 7 iOS project with a simple Swift file

### ISSUE THREAD

**Issue by lemonkey** [gianlucabertani/Objective-Zip#39]
**cocopods v0.39, use_frameworks!, objective-zip v1.0.2**

Just by importing the objective_zip module and nothing else via `import objective_zip`, I get the following compiler errors:

   Pods/objective-zip/ZLib/crc32.h:5:1: Unknown type name 'local'
   Pods/objective-zip/ZLib/crc32.h:5:7: Expected identifier or '('
   Pods/objective-zip/ZLib/deflate.h:97:16: Redefinition of 'internal_state'
   Pods/objective-zip/ZLib/inffixed.h:10:18: Unknown type name 'code'
   Pods/objective-zip/ZLib/inffixed.h:87:18: Unknown type name 'code'
   Pods/objective-zip/ZLib/inflate.h:36:9: Expected identifier
   Pods/objective-zip/ZLib/inflate.h:106:5: Unknown type name 'code'
   Pods/objective-zip/ZLib/inflate.h:106:10: Expected member name or ';' after declaration specifiers
   Pods/objective-zip/ZLib/inflate.h:106:9: Expected ';' at end of declaration list
   Pods/objective-zip/ZLib/inflate.h:107:5: Unknown type name 'code'
   Pods/objective-zip/ZLib/inflate.h:107:10: Expected member name or ';' after declaration specifiers
   Pods/objective-zip/ZLib/inflate.h:107:9: Expected ';' at end of declaration list
   Pods/objective-zip/ZLib/inflate.h:115:5: Unknown type name 'code'
   Pods/objective-zip/ZLib/inflate.h:118:5: Unknown type name 'code'
   Pods/objective-zip/ZLib/inflate.h:118:16: Use of undeclared identifier 'ENOUGH'
   Pods/objective-zip/ZLib/trees.h:73:36: Use of undeclared identifier 'DIST_CODE_LEN'
   SampleSwift.swift:41:8: Could not build Objective-C module 'objective_zip'

**Comment by gianlucabertani** [comment]
You should set a bridging header, since Objective-Zip is actually a mixed C/Objective-C project. Take a look under the `Objective-Zip Tests` folder, you can find an example of bridging header and a sample Swift source code with a some use cases.

**Comment by lemonkey** [comment]
Thanks!

**Comment by lemonkey** [comment]
I think the problem is that the Objective-C umbrella header that is generated by CocoaPods while integrating the framework with use_frameworks! includes both the Objective-C and C headers. This is because both Objective-C and C headers have public framework target membership.

When I import the framework in a sample Swift file using `import objective_zip`, it fails on the C class headers since they're in the framework's umbrella header.

Within the same app I can create an Objective-C bridging header and include only those Objective-C class headers in the app that I want to be available to my app's Swift classes, but it was my understanding that across frameworks you only use `import framework_name`.

See "exporting external frameworks" section:

https://developer.apple.com/library/ios/documentation/Swift/Conceptual/BuildingCocoaApps/MixandMatch.html#//apple_ref/doc/uid/TP40014216-CH10-ID122

**Comment by lemonkey** [comment]
I wonder if it's an issue specific to the fact that I have objective-zip as a dependency to my Cocoa Touch Framework which is a dependency to my iOS app project. When pod install builds the Pods project, the objective-zip target compile sources listing shows -DOS_OBJECT_USE_OBJC=0 for the C files, but when my iOS app tries to build my framework that has objective-zip as a dependency, in the framework files that import "Objective-Zip.h", it causes the compiler to fail with the errors originally mentioned in my first post..

**Comment by lemonkey** [comment]
The first compiler error I see is the following:

In file included from /iphone_test/Pods/Target Support Files/objective-zip/objective-zip-umbrella.h:26:
/iphone_test/Pods/objective-zip/ZLib/crc32.h:5:1: error: unknown type name 'local'
local const z_crc_t FAR crc_table[TBLS][256] =

This doesn't happen when I build my framework on its own, but it does when I build the iOS app that uses the framework... I'm stumped.

**Comment by lemonkey** [comment]
I can also make the compile fail in just my test framework by using @import objective_zip; as opposed to #import "Objective-Zip.h"

**Comment by lemonkey** [comment]
The problem is that crc32.h is the output of a previous compilation that dynamically generated the header file, however it references keywords that are only defined in crc32.c. This is preventing the objective_zip module from compiling as crc32.h is included in the module map by way of the umbrella header.

**Comment by lemonkey** [comment]
What I can't understand is how #import "Objective-Zip.h" doesn't lead to this same compiler error when @import objective_zip; is used instead... Both zip.h and unzip.h are being included by OZZipFile.h

**Comment by lemonkey** [comment]
If I exclude the C files from the module map, it everything compiles, but I doubt the zip/unzip functionality will work in the context of the module...

**Comment by lemonkey** [comment]
After researching modules a bit further, I've found that one of the selling points of modules is that they're immune to the effects of local macro definitions:

http://stackoverflow.com/a/18947634

And it turns out that this is exactly what crc32.h is relying upon, in that the macros that allow all of the C code in Objective-Zip to compile successfully are defined in crc32.c.

As the Objective-Zip pod is integrating into my framework via use_frameworks! and since we have the entire source (as opposed to a "fake" framework being created with just a static binary and some headers), it is successfully being turned into a dynamic framework, where **modules are enabled** and a module map is being generated via the framework header membership levels (public, project, etc.). Once you try to use @import objective_zip or the dynamic framework you're using a user header import `#import "Objective-Zip.h"` with is integrated into another project, the module for Objective-Zip will attempt to compile and fail.

Long story short, CLANG_ENABLE_MODULES needed to be disabled for the Objective-Zip CocoaPod when integrating it with use_frameworks! in the Podfile via a post_install step.

**Comment by gianlucabertani** [comment]
Hopefully things will just work out of the box once I will have updated the project to a framework. My guess is, in fact, that if the project was a framework header files like crc32.h would not even be imported, since they would be private.

Thank you very much for your investigation.

**Comment by lemonkey** [comment]
That was my initial thought too, but after removing those headers from the dynamically generated umbrella header and module map files that CocoaPods created it failed to compile. The ObjC classes import zip.h and unzip.h. The main problem are the local define macros in crc32.c and how modules ignore them.

**Comment by eranmarom** [comment]
I am having the same issues, and I could not understand what should be the fix. Can you please elaborate.

**Comment by lemonkey** [comment]
@eranmarom what I ended up doing late last year was to update my Podfile to include a post_install step that disables modules for objective_zip specifically. I haven't had a chance to re-evaluate any newer version of this library lately.

**Comment by eranmarom** [comment]
I ended up using a different library which gave me no problems with dynamic frameworks and pods: https://github.com/ZipArchive/ZipArchive
Thanks anyway.
